code/datums/chatmessage.dm

```diff
@@ -140,7 +140,7 @@
 	approx_lines = max(1, mheight / CHAT_MESSAGE_APPROX_LHEIGHT)
 
 	// Translate any existing messages upwards, apply exponential decay factors to timers
-	message_loc = target
+	message_loc = isturf(target) ? target : get_atom_on_turf(target)
```

Comment:
I am a bit scared by that while loop on every message tbh

Reply:
We use it in every balloon message already /shrug

Reply:
And well it does do what it says